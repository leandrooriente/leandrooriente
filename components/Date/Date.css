.details {
  color: var(--secondary-color);
  font-family: var(--headline-font-family);
  font-size: 14px;
  text-transform: capitalize;
  font-style: italic;
}

.separator {
  color: var(--light-gray);
  font-style: normal;
  font-weight: bold;
  display: inline-block;
  margin: 0 5px;
}