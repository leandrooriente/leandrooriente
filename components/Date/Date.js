import React from 'react';
import PropTypes from 'prop-types';
import formatArticleDate from 'helpers/formatArticleDate';
import css from './Date.css';

const Date = ({ date, lang, className }) => {
  const classes = `${css.details}${className ? ` ${className}` : ''}`;

  return (
    <div className={classes}>
      {formatArticleDate(date)} <span className={css.separator}>-</span> {lang}
    </div>
  );
};

Date.defaultProps = {
  className: null,
};

Date.propTypes = {
  className: PropTypes.string,
  date: PropTypes.string.isRequired,
  lang: PropTypes.string.isRequired,
};

export default Date;
