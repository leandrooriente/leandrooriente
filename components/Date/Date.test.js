import React from 'react';
import { mount } from 'enzyme';
import Date from './Date';

describe('<Date />', () => {
  it('should render <Date /> component', () => {
    const wrapper = mount(
      <Date
        lang="portuguese"
        date="10/10/2010"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
