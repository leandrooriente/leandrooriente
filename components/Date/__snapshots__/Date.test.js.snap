// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Date /> should render <Date /> component 1`] = `
<Date
  className={null}
  date="10/10/2010"
  lang="portuguese"
>
  <div
    className="details"
  >
    October 2010
     
    <span
      className="separator"
    >
      -
    </span>
     
    portuguese
  </div>
</Date>
`;
